frontend/compiled/cs.go

```diff
@@ -34,12 +41,17 @@ type ConstraintSystem struct {
 	// debug info contains stack trace (including line number) of a call to a cs.API that
 	// results in an unsolved constraint
 	DebugInfo []LogEntry
+	// maps unique id to a path (optimized for reading debug info stacks)
+	DebugStackPaths map[uint32]string
```

Comment:
Hmm. Would it make sense to prevent serializing `DebugInfo`, `Logs` like `debugPathsIds` and `debugPathId`? I think the use-case for debug info is debugging the circuit locally and then we can assume that we haven't had serialized->deserialized step?

Reply:
No, `DebugInfo` and `Logs` can be used after deserialization; say you have a prover service, and it solves with invalid witness, these 2 objects will help produce a readable error. 

+ logs is for the `api.Println` that you may wish to log or what not.